plugins/woocommerce/src/Internal/ProductFilters/TaxonomyHierarchyData.php

```
@@ -93,29 +143,69 @@ private function build_full_hierarchy_map( string $taxonomy ): array {
 
+	/**
+	 * Recursively build hierarchical term tree with depth and parent.
+	 *
+	 * @param array $tree       Reference to tree array being built.
+	 * @param int   $term_id    Current term ID.
+	 * @param array $children   Children relationships map (parent_id => [child_ids]).
+	 * @param array $temp_terms Term data indexed by term_id.
+	 * @param int   $parent_id  Current parent term ID.
+	 * @param int   $depth      Current depth level in hierarchy.
+	 */
+	private function build_term_tree( &$tree, $term_id, $children, $temp_terms, $parent_id = 0, $depth = 0 ) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$parent_id'. (Unused Code Rules)

(UnusedFormalParameter)